src/network/network_query.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file network_query.h Query part of the network protocol. */
+
+#ifndef NETWORK_QUERY_H
+#define NETWORK_QUERY_H
+
+#include "network_internal.h"
+
+/** Class for handling the client side of quering a game server. */
+class QueryNetworkGameSocketHandler : public ZeroedMemoryAllocator, public NetworkGameSocketHandler {
```

Comment:
Is the `ZeroedMemoryAllocator` really needed here? Only the `ClientNetworkGameSocketHandler` uses it but not the `ServerNetworkGameSocketHandler` so it is not something the `NetworkGameSocketHandler` intrinsicly needs. Since this has only one instance variable that gets in the constructor, I think there is no use for it.

Reply:
`ServerNetworkGameSocketHandler` also zero's the object, but this is done by the pool (`Tzero` is set to `true`). So there too `Calloc` is used. In other words: it currently is consistent in zero'ing the object on creation.

This is why I left it in, as I was unsure what the side-effects are of not doing it. `NetworkGameSocketHandler` has at least one variable that is otherwise not set.

Of course these days we can just set the initial values of variables, and possibly get ride of this whole `ZeroedMemoryAllocator`, but that might be best for another PR. For now, I suggest we leave it consistent, and zero the object for all children using `NetworkGameSocketHandler`.